app/backend/app.py

```diff
@@ -43,6 +44,9 @@
 openai_token = azure_credential.get_token("https://cognitiveservices.azure.com/.default")
 openai.api_key = openai_token.token
 
+# Create chatGptProxy
+chatgpt_proxy = ChatGptProxy(openai, AZURE_OPENAI_CHATGPT_DEPLOYMENT, AZURE_OPENAI_CHATGPT_MODEL, AZURE_OPENAI_GPT_DEPLOYMENT)
```

Comment:
Hm, It seems that you're passing the entire openai module here, but then also importing openai in the chatgptproxy file. I need to research more, but I'm not sure whether that's a good idea, it may lead to some strangeness.

Ideally, the OpenAI SDK would actually allow the construction of an OpenAI() object with the credentials, instead of us having to set the credentials as globals on the module. With the current OpenAI SDK design, we have to be very careful about where we set credentials and the way  we import openai.

Reply:
Thanks Pamlea. I shared my thoughts for the changes in above comments. I have removed the proxy and kept implementation simple as it was.